src/components/Coaching/CoachingDetail/Activity/Activity.tsx

```diff
@@ -194,21 +194,32 @@ export const Activity: React.FC<ActivityProps> = ({
   const { t } = useTranslation();
   const locale = useLocale();
 
-  const periodUnit: DateTimeUnit =
-    period === CoachingPeriodEnum.Weekly ? 'week' : 'month';
   const periodDuration =
     period === CoachingPeriodEnum.Weekly ? { weeks: 1 } : { months: 1 };
-  const [start, setStart] = useState(DateTime.now().startOf(periodUnit));
-  const end = useMemo(() => start.endOf(periodUnit), [start, periodUnit]);
+  const startDate = useMemo((): DateTime => {
+    // The default startOf('week') is Monday. Subtract 1 day to start the week on Sunday.
+    if (period === CoachingPeriodEnum.Weekly) {
+      return DateTime.now().startOf('week').minus({ day: 1 });
+    }
+    return DateTime.now().startOf('month');
+  }, [period]);
+  const [start, setStart] = useState(startDate);
+  const end = useMemo(
+    () =>
+      period === CoachingPeriodEnum.Weekly
+        ? start.plus({ days: 6 }).endOf('day')
```

Comment:
Do you think that `start.endOf('week').minus({ day: 1 })` would be clearer to match line 202?

Reply:
`start.endOf('week')` returns the same day as `start`(sunday) because it considers sunday the end of the week. So the end of the week of the day sunday is the same day, sunday. And then we subtract 1 day and we get `Oct 6 - 5`